cg/services/orders/store_order_services/store_case_order.py

```diff
@@ -34,263 +38,182 @@ def __init__(
-            if data_analysis in [Workflow.MIP_DNA, Workflow.TOMTE]:
-                panels: set[str] = {
-                    panel for sample in case_samples for panel in sample.panels if panel
-                }
-
-            priority = self._get_single_value(
-                case_name, case_samples, "priority", Priority.standard.name
-            )
-            synopsis: str = self._get_single_value(case_name, case_samples, "synopsis")
-
-            case = {
-                "cohorts": list(cohorts),
-                "data_analysis": data_analysis,
-                "data_delivery": data_delivery,
-                "internal_id": case_internal_id,
-                "name": case_name,
-                "panels": list(panels),
-                "priority": priority,
-                "samples": [
-                    {
-                        "age_at_sampling": sample.age_at_sampling,
-                        "application": sample.application,
-                        "capture_kit": sample.capture_kit,
-                        "comment": sample.comment,
-                        "control": sample.control,
-                        "father": sample.father,
-                        "internal_id": sample.internal_id,
-                        "mother": sample.mother,
-                        "name": sample.name,
-                        "phenotype_groups": list(sample.phenotype_groups),
-                        "phenotype_terms": list(sample.phenotype_terms),
-                        "reference_genome": (
-                            sample.reference_genome if hasattr(sample, "reference_genome") else None
-                        ),
-                        "sex": sample.sex,
-                        "status": sample.status if hasattr(sample, "status") else None,
-                        "subject_id": sample.subject_id,
-                        "tumour": sample.tumour,
-                    }
-                    for sample in case_samples
-                ],
-                "synopsis": synopsis,
-            }
-
-            status_data["families"].append(case)
-        return status_data
-
-    def store_items_in_status(
-        self, customer_id: str, order: str, ordered: datetime, ticket_id: str, items: list[dict]
-    ) -> list[Case]:
+    def store_items_in_status(self, order: OrderWithCases) -> list[DbCase]:
         """Store cases, samples and their relationship in the Status database."""
         customer: Customer = self.status_db.get_customer_by_internal_id(
-            customer_internal_id=customer_id
+            customer_internal_id=order.customer
         )
-        new_cases: list[Case] = []
-        status_db_order = Order(
-            customer=customer,
-            order_date=datetime.now(),
-            ticket_id=int(ticket_id),
+        new_cases: list[DbCase] = []
+        status_db_order = DbOrder(
+            customer=customer, order_date=datetime.now(), ticket_id=order._generated_ticket_id
```

Comment:
Refactoring: consider wrapping this into `self._create_db_order` as in the fastq store service